### ISSUE TITLE
Does not inject custom JNDI resource of type Integer

### ISSUE THREAD

**Issue by c-kunz** [payara/Payara#3935]
# Description #

Injecting a JNDI custom resource of type `java.lang.Integer` should work as advertised here:
https://blog.payara.fish/building-your-next-microservice-with-eclipse-microprofile
https://docs.payara.fish/documentation/microprofile/config.html

## Expected Outcome

Deploys the application and returns the value of the custom resource property `test` of type `java.lang.Integer`.

## Current Outcome

Does not deploy the application.

Stack trace (top-most exception):

```
javax.enterprise.inject.spi.DeploymentException: Deploment Failure for ConfigProperty test in class de.quipsy.kunz.test.Resource Reason Microprofile Config Property test can not be found
```

## Steps to reproduce

1. `git clone https://github.com/c-kunz/mp-config-payara-bug`
2. `cd mp-config-payara-bug`
3. `mvn package`
4. Create custom resource of type java.lang.Integer
`asadmin create-custom-resource --restype java.lang.Integer --factoryclass org.glassfish.resources.custom.factory.PrimitivesAndStringFactory --property value=123 test`
5. `cd target`
6. `asadmin deploy demo.war`

## Hint
Injecting `payara.instance.http.port` instead of property `test` DOES WORK.
Setting `--restype` of property to `java.lang.String` and injecting using
```java
@Inject
@ConfigProperty
private String test;
```
DOES WORK as expected.

### Samples
https://github.com/c-kunz/mp-config-payara-bug

## Environment ##

- **Payara Version**: 4.1.2.181
- **Edition**: Full
- **JDK Version**: 8u192 - Zulu
- **Operating System**: Linux on Ubuntu 18.04

**Comment by rdebusscher** [comment]
Hi Christian,

Thank you for the report of this issue.

I can confirm the behavior with Integer and String. It is also reproducible with Payara 5.191. Therefore I have created the internal issue `PAYARA-3796` to fix this.

Be aware that it will only be fixed in Payara 5 (not 4)

Regards
Rudy

**Comment by rdebusscher** [comment]
Hi,

This is fixed in the release 5.194.
